### DISCUSSION TITLE
Introduction of Custom Number Formatting - Pivot Table v2

### DISCUSSION THREAD

**Discussion by Wojtek17** [apache/superset#24795] [Ideas]
Hey @kgabryje, I have a suggestion. Wouldn't it be good to introduce custom number formatting? The options we have sometimes aren't enough. For my specifc case I urgently need format so that I have decimals separeted with comma and thousands with a space. Where can I dig in the code to create such option? Thanks in advance.

**Comment by kgabryje** [comment]
`D3_FORMAT` in the config file controls locale specific formatting such as decimal and thousands separators. Keep in mind though that that setting is global - I don't think we have an option to override locale per chart. CC @yousoph
Check out the [docs](https://superset.apache.org/docs/installation/configuring-superset/#configuring-superset) to see how to override default configuration

**Reply by kgabryje** [reply]
`D3_FORMAT: D3Format = {"decimal": ",", "thousands": " "}` should probably work for you

**Reply by Wojtek17** [reply]
Hey, I don't think it's the best option either. I don't want to change this global setting but rather add option in here:
![image](https://github.com/apache/superset/assets/112469140/655900a2-b967-4aa9-85e8-7649c3ff794f)
Then I could use it on daily basis. Is there a way to do it? I've searched for these options but didn't find any of these formats in code

**Reply by Wojtek17** [reply]
I think I found it here: https://github.com/apache/superset/blob/fc89718d48c9aa034131a35d5ccc7cefc6f9ddef/superset-frontend/src/explore/controls.jsx#L88

**Reply by kgabryje** [reply]
D3 format strings still use locale options defined in D3_FORMAT (https://github.com/d3/d3-format), so unfortunately what you're looking for is basically a new feature request.
